app/services/user/bulk_destroy.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class User::BulkDestroy
+  include Service::Base
+
+  params do
+    attribute :user_ids, :array
+
+    validates :user_ids, length: { maximum: 100 }
+  end
+
+  model :users
+  policy :can_delete_users
+  step :delete
+
+  private
+
+  def fetch_users(params:)
+    ids = params.user_ids
+    # this order cluase ensures we retrieve the users in the same order as the
+    # IDs in the param. we do this to ensure the users are deleted in the same
+    # order as they're selected in the UI
```

Comment:
That's cool, but is it something we need to care about in the first place? 🙂

Reply:
A couple of reason I did this:

1. when I was testing locally, I got a bit confused to see the users being deleted (in the progress logs) in a completely different order to how I selected them
2. in the system specs for the progress logs UI, I need to assert that for each deleted user, we get an entry in the logs. An entry reads like `[X/Y] Successfully deleted @abc` where X is the user position in the list and Y is the total number of users. If the users were retrieved in a random order, the X would be different in each test run.